src/index.ts

```diff
@@ -164,15 +183,54 @@ export class Storage {
       }
 
       for (const key of relevantKeyList) {
-        callbackMap[key]?.(
-          {
-            newValue: this.#parseValue(changes[key].newValue),
-            oldValue: this.#parseValue(changes[key].oldValue)
-          },
-          areaName
+        this.#watchListeners[key]?.forEach((callback) =>
+          callback(
+            {
+              newValue: this.#parseValue(changes[key].newValue),
+              oldValue: this.#parseValue(changes[key].oldValue)
+            },
+            areaName
+          )
         )
       }
-    })
+    }
+
+    chrome.storage.onChanged.addListener(this.#internalStorageListener)
+  }
+
+  unwatch = (callbackMap: Record<string, StorageWatchListener>): boolean => {
+    if (!this.isWatchingSupported()) return false
+
+    for (const [key, callback] of Object.entries(callbackMap)) {
```

Comment:
I think this section can use map instead of a forloop, which would allow chaining with filter